lib/sisimai/bite/json/amazonses.rb

```diff
@@ -0,0 +1,255 @@
+      #                             arguments are missing
+      # @since v4.20.0
+      def adapt(argvs)
+        return nil unless argvs.is_a? Hash
+        return nil unless argvs.keys.size > 0
+        return nil unless argvs.key?('notificationType')
+
+        require 'sisimai/rfc5322'
+        dscontents = [Sisimai::Bite.DELIVERYSTATUS]
+        rfc822head = {}   # (Hash) Check flags for headers in RFC822 part
+        recipients = 0    # (Integer) The number of 'Final-Recipient' header
+        labeltable = {
+          :Bounce    => 'bouncedRecipients',
+          :Complaint => 'complainedRecipients',
+        }
+        v = nil
+
+        if argvs['notificationType'] =~ /\A(?:Bounce|Complaint)\z/
+          # { "notificationType":"Bounce", "bounce": { "bounceType":"Permanent",...
+          o = argvs[ argvs['notificationType'].downcase ]
+          r = o[ labeltable[ argvs['notificationType'].to_sym ] ] || []
+
+          r.each do |e|
+            # 'bouncedRecipients' => [ { 'emailAddress' => 'bounce@si...' }, ... ]
+            # 'complainedRecipients' => [ { 'emailAddress' => 'complaint@si...' }, ... ]
+            next unless Sisimai::RFC5322.is_emailaddress(e['emailAddress'])
+
+            v = dscontents[-1]
+            if v['recipient']
+              # There are multiple recipient addresses in the message body.
+              dscontents << Sisimai::Bite.DELIVERYSTATUS
+              v = dscontents[-1]
+            end
+            recipients += 1
+            v['recipient'] = e['emailAddress']
+
+            if argvs['notificationType'] == 'Bounce'
+              # 'bouncedRecipients => [ {
+              #   'emailAddress' => 'bounce@simulator.amazonses.com',
+              #   'action' => 'failed',
+              #   'status' => '5.1.1',
+              #   'diagnosticCode' => 'smtp; 550 5.1.1 user unknown'
+              # }, ... ]
+              v['action'] = e['action']
+              v['status'] = e['status']
+
+              if cv = e['diagnosticCode'].match(/\A(.+?);[ ]*(.+)\z/)
+                # Diagnostic-Code: SMTP; 550 5.1.1 <userunknown@example.jp>... User Unknown
+                v['spec'] = cv[1].upcase
+                v['diagnosis'] = cv[2]
+              else
+                v['diagnosis'] = e['diagnosticCode']
+              end
+
+              if cv = o['reportingMTA'].match(/\Adsn;[ ](.+)\z/)
+                # 'reportingMTA' => 'dsn; a27-23.smtp-out.us-west-2.amazonses.com',
+                v['lhost'] = cv[1]
+              end
+
+              if BounceType.key?(o['bounceType'].to_sym) &&
+                 BounceType[o['bounceType'].to_sym].key?(o['bounceSubType'].to_sym)
+                  # 'bounce' => {
+                  #       'bounceType' => 'Permanent',
+                  #       'bounceSubType' => 'General'
+                  # },
+                  v['reason'] = BounceType[o['bounceType'].to_sym][o['bounceSubType'].to_sym]
+              end
```

Comment:
__[rubocop]__
* Layout/CommentIndentation: Incorrect indentation detected (column 18 instead of 16).

<!-- Comment by [SideCI](https://sideci.com) -->